project/dexter-metrics/src/java/com/samsung/sec/dexter/metrics/util/CdtUtil.java

```diff
@@ -0,0 +1,169 @@
+/**
+ * Copyright (c) 2014 Samsung Electronics, Inc.,
+ * All rights reserved.
+ * 
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ * 
+ * * Redistributions of source code must retain the above copyright notice, this
+ *   list of conditions and the following disclaimer.
+ * 
+ * * Redistributions in binary form must reproduce the above copyright notice,
+ *   this list of conditions and the following disclaimer in the documentation
+ *   and/or other materials provided with the distribution.
+ * 
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE
+ * FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+ * DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+ * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+package com.samsung.sec.dexter.metrics.util;
+
+import java.nio.charset.Charset;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.List;
+
+import org.apache.log4j.Logger;
+import org.eclipse.cdt.core.dom.ast.ASTVisitor;
+import org.eclipse.cdt.core.dom.ast.IASTDeclaration;
+import org.eclipse.cdt.core.dom.ast.IASTTranslationUnit;
+import org.eclipse.cdt.core.parser.ParserLanguage;
+
+import com.google.common.base.Charsets;
+import com.samsung.sec.dexter.util.TranslationUnitFactory;
+
+public class CdtUtil {
+	static Charset sourceEncoding = Charsets.UTF_8;
+	static Logger logger = Logger.getLogger(CdtUtil.class);
+
+	/**
+	 * ExtractModuleName(String sourceFilePath, final int lineNumber) method is
+	 * responsible for getting ModuleInformation
+	 * 
+	 * @param [in] String sourceFilePath, final int lineNumber
+	 * @return [out] Map<String, String>
+	 * @warning [None]
+	 * @exception IO
+	 *                exception
+	 */
+	public static synchronized Map<String, String> extractModuleName( String sourceFilePath, final int lineNumber) {
+		Map<String, String> mapModuleName = null;
+
+		String code = CdtUtilHelper.getContentsFromFile(sourceFilePath, sourceEncoding);
+		IASTTranslationUnit translationUnit = TranslationUnitFactory.getASTTranslationUnit(code, ParserLanguage.CPP, 
+				sourceFilePath);
+
+		final String fileExtension = sourceFilePath.substring(sourceFilePath.indexOf('.'));
```

Comment:
A file name or path can have '.' in the middle of the name/path: a.b.c  a/b.c/d.c
You have to check the '.' from end side.
Please use utility lib or make it as utility in DexterUtil.java